projects/plugins/jetpack/tests/php/sync/server/class.jetpack-sync-server-eventstore.php

```diff
@@ -24,30 +24,48 @@ public function handle_remote_action( $action_name, $args, $user_id, $silent, $t
 		);
 	}
 
-	public function get_all_events( $action_name = null, $blog_id = null ) {
+	/**
+	 * Returns all Sync events of a certain action type, of a certain blog, and filtered if necessary.
+	 *
+	 * @param String   $action_name Sync action slug, e.g. jetpack_sync_save_post.
+	 * @param Integer  $blog_id Blog ID filter - only return events for a given blog ID, defaults to current blog.
+	 * @param Callable $filter a custom callable to pass the event object to to be filtered.
+	 **/
+	public function get_all_events( $action_name = null, $blog_id = null, $filter = null ) {
 		$blog_id = isset( $blog_id ) ? $blog_id : get_current_blog_id();
 
 		if ( ! isset( $this->events[ $blog_id ] ) ) {
 			return array();
 		}
 
-		if ( $action_name ) {
-			$events = array();
+		$events = array();
 
+		if ( $action_name ) {
 			foreach ( $this->events[ $blog_id ] as $event ) {
 				if ( $event->action === $action_name ) {
 					$events[] = $event;
 				}
 			}
+		} else {
+			$events = $this->events[ $blog_id ];
+		}
 
-			return $events;
+		if ( is_callable( $filter ) ) {
+			$events = array_filter( $events, $filter );
```

Comment:
Looks like tests are failing because this leaves holes in the array, which callers aren't expecting.
```suggestion
			$events = array_values( array_filter( $events, $filter ) );
```

Reply:
Whoops, broke more tests than fixed :)